### DISCUSSION TITLE
How can add session on easy api?

### DISCUSSION THREAD

**Discussion by shayanfh** [open-wa/wa-automate-nodejs#2550] [Q&A]
I need something this `http://localhost:8080/addSession?id=s2` to add session with api .
now we should first add all session first with `npx @open-wa/wa-automate -p 8080 -s s1`  ,then use them with api .
Actually, i need to use wa-automated-nodejs on backend web and another language programming.

**Comment by smashah** [comment]
You will need to build this out on your own as the official open-wa solution is for clients only.

Essentially, there are 2/3 parts to this.

1. Process management
2. API layer for process management
3. [optional] reverse proxy

Depending on your resources, you can do this multiple ways.

You can do it via kubernetes, quite difficult but robust solution. [Spawner](https://github.com/drifting-in-space/spawner) may make this easier but it's not completely fit for purpose mainly because Sweeper cleans up pods with stale connections 

Or you can develop a custom solution on top of docker using the docker image [`docker run openwa/wa-automate`](https://hub.docker.com/repository/docker/openwa/wa-automate)

If you don't want to deal with container system (there are good reasons to avoid a dockerized system), you can use a node-only type setup where for 1 you have [`pm2`](https://pm2.io/) for 2 you can use something like [`rpm2`](https://www.npmjs.com/package/rpm2) (unmaintained, you will need to fork and fix it)

Another option is to fork https://github.com/joaomirandasa/whats2api and just update it as it was developed on top of @open-wa/wa-automate from the start.

**Reply by shayanfh** [reply]
We have a project,
can we contract with you?

**Reply by smashah** [reply]
@shayanfh sure just email me at shah@openwa.dev
